scripts/service-principals.sh

```diff
@@ -0,0 +1,29 @@
+set -ex
+
+CHECK_DAYS=$1
+
+TODAY_DATE=$(date +%Y-%m-%d)
+
+CHECK_DATE=$(date -d "+${CHECK_DAYS} days" +%Y-%m-%d)
+
+
+AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '${TODAY_DATE}' && endDateTime < '${CHECK_DATE}']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[*].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+#AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-10-14']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[*].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+AZ_APP_COUNT=$(jq -r '. | length' <<< "${AZ_APP_RESULT}")
+
+if [[ $AZ_APP_COUNT == 0 ]]; then
+    echo "None Service Principals Secrets are expiring in $CHECK_DAYS days"
```

Comment:
we should still append slack message in this case with green light.

Reply:
done